File: src/main/Java/boj_step16/boj18258.java
import java.io.*;
import java.util.LinkedList;
import java.util.StringTokenizer;

public class boj18258 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        int n = Integer.parseInt(reader.readLine());
        LinkedList<Integer> queue = new LinkedList<>();
        while (n-- > 0) {
            StringTokenizer tokenizer = new StringTokenizer(reader.readLine());
            switch (tokenizer.nextToken()) {
                case "push": queue.offer(Integer.parseInt(tokenizer.nextToken())); break;
                case "pop":
                    if (queue.isEmpty()) writer.append("-1\n");
                    else writer.append(queue.poll() + "\n"); break;
                case "size": writer.append(queue.size() + "\n"); break;
                case "empty": writer.append((queue.isEmpty()) ? "1\n" : "0\n"); break;
                case "front":
                    if (queue.isEmpty()) writer.append("-1\n");
                    else writer.append(queue.peek() + "\n"); break;
                case "back":
                    if (queue.isEmpty()) writer.append("-1\n");
                    else writer.append(queue.peekLast() + "\n");
            }
        }
        writer.close();
    }
}

File: src/main/Kotlin/boj_step16/boj18258.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.*

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    val n = r.readLine().toInt()
    val queue = LinkedList<Int>()
    repeat(n) {
        val command = r.readLine()
        val token = StringTokenizer(command)
        when (token.nextToken().toString()) {
            "push" -> queue.add(token.nextToken().toInt())
            "pop" -> {
                if (queue.isEmpty()) w.write("-1\n")
                else w.write(queue.removeFirst().toString() + "\n")
            }
            "size" -> w.write(queue.size.toString() + "\n")
            "empty" -> w.write(queue.isEmpty().compareTo(false).toString() + "\n")
            "front" -> {
                if (queue.isEmpty()) w.write("-1\n")
                else w.write(queue.first().toString() + "\n")
            }
            "back" -> {
                if (queue.isEmpty()) w.write("-1\n")
                else w.write(queue.last().toString() + "\n")
            }
        }
    }
    w.close()
}